filip/clients/ngsi_v2/cb.py

```diff
@@ -601,32 +592,159 @@ def delete_entities(self, entities: List[ContextEntity]) -> None:
         if len(entities_with_attributes) > 0:
             self.update(entities=entities_with_attributes, action_type="delete")
 
-    def replace_entity_attributes(self,
-                                  entity: ContextEntity,
-                                  options: str = None,
-                                  append: bool = True):
+    def update_or_append_entity_attributes(
+            self,
+            entity_id: str,
+            entity_type: str,
+            attrs: List[Union[NamedContextAttribute,
+                              Dict[str, ContextAttribute]]],
+            append: bool = False):
         """
-        The attributes previously existing in the entity are removed and
-        replaced by the ones in the request.
+        The request payload is an object representing the attributes to
+        append or update. This corresponds to a 'POST' request if append is
+        set to 'False'
+
+        Note:
+            Be careful not to update attributes that are
+            provided via context registration, e.g. commands. Commands are
+            removed before sending the request. To avoid breaking things.
 
         Args:
-            entity (ContextEntity):
-            append (bool):
-            options:
+            entity_id: Entity id to be updated
+            entity_type: Entity type, to avoid ambiguity in case there are
+                several entities with the same entity id.
+            attrs: List of attributes to update or to append
+            append: If `False` the entity attributes are updated (if they
```

Comment:
I find the "append" flag quite confusing. 
In summary: if true non-existing are appended, and also if false they are also appended. 
The only difference is with updating ?

Reply:
i noticed that but simply used it from the specs. I changed it to `append_strict`.